### PR TITLE
ADD tests for OpenShift GitOps

### PR DESCRIPTION
**Pull request by acmenezes**
- Directory name matching operator's package name
- Test namespace creation
- ArgoCD creation until Available
- Sample application creation until Synced
- Clean Up tasks added

### CODE REVIEW

File: openshift-gitops-operator/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      retries: 10
+      delay: 10
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Error testing argocd with application. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "failed"
+- name: Clean Up resources
```
Comment (madorn) [inline openshift-gitops-operator/tasks/main.yml:150]: @acmenezes should we add a task to cleanup the `openshift-gitops` namespace? it sticks around, even when the operator is uninstalled.

Comment (acmenezes) [reply openshift-gitops-operator/tasks/main.yml:150]: @madorn , second take on it. After investigation I can only conclude that something else wen't wrong when I tried to delete it. Now the code is deleting openshift-gitops and completing well all tasks. So I guess we may merge this one. Let me know what you think.

File: openshift-gitops-operator/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        state: present
+    # 2. Testing argocd Instance
+    - name: Create argocd
+      kubernetes.core.k8s:
+        name: argocd
+        namespace: test-gitops
+        api_version: argoproj.io/v1alpha1
+        kind: ArgoCD
+        definition:
+          metadata:
```
Comment (bcrochet) [inline openshift-gitops-operator/tasks/main.yml:22]: Given the size of this, I'd put it in a template.

File: openshift-gitops-operator/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,191 @@
+---
+- name: Setting Up Results Var
+  ansible.builtin.set_fact:
+    test_result: "pass"
```
Comment (bcrochet) [inline openshift-gitops-operator/tasks/main.yml:4]:
```suggestion
    assessment_result: "pass"
```

File: openshift-gitops-operator/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      until: app.resources[0].status.sync.status is defined and app.resources[0].status.sync.status == "Synced"
+      retries: 10
+      delay: 10
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Error testing argocd with application. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "failed"
```
Comment (bcrochet) [inline openshift-gitops-operator/tasks/main.yml:149]:
```suggestion
        assessment_result: "fail"
```

File: openshift-gitops-operator/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        state: absent
+        wait: true        
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Error cleaning up argocd with application. Err: {{ ansible_failed_result }}"
+# print pass or failed according to test_result fact variable
+- name: Openshift-GitOps Results
+  ansible.builtin.debug:
+    msg: "{% if test_result == 'pass' %} PASS {% else %} FAILED {% endif %}"
```
Comment (bcrochet) [inline openshift-gitops-operator/tasks/main.yml:191]:
```suggestion
    msg: "{{ (assessment_result == 'pass') | ternary("PASS", "FAIL") }}"
```

**Review by bcrochet — CHANGES_REQUESTED** [review]
Suggestions on using assessment_result directly

File: openshift-gitops-operator/tasks/main.yml (hunk 2 above)
Comment (acmenezes) [reply openshift-gitops-operator/tasks/main.yml]: Done.

File: openshift-gitops-operator/tasks/main.yml (hunk 3 above)
Comment (acmenezes) [reply openshift-gitops-operator/tasks/main.yml]: Replaced by a single report result task.

File: openshift-gitops-operator/tasks/main.yml (hunk 4 above)
Comment (acmenezes) [reply openshift-gitops-operator/tasks/main.yml]: replaced by a single result report task

File: openshift-gitops-operator/tasks/main.yml (hunk 5 above)
Comment (acmenezes) [reply openshift-gitops-operator/tasks/main.yml]: No longer needed. Check the single report results task.

File: openshift-gitops-operator/templates/argocd.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                - TaskRun
+                - PipelineRun
+            controller:
+              resources:
+                limits:
+                  cpu: 2000m
+                  memory: 2048Mi
+                requests:
+                  cpu: 250m
+                  memory: 1024Mi
```
Comment (bcrochet) [inline openshift-gitops-operator/templates/argocd.yml.j2:71]: Needs a newline.

File: openshift-gitops-operator/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      kubernetes.core.k8s:
+        name: openshift-gitops
+        api_version: v1
+        kind: Namespace
+        state: absent
+        wait: true        
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "Error cleaning up argocd with application. Err: {{ ansible_failed_result }}"
```
Comment (bcrochet) [inline openshift-gitops-operator/tasks/main.yml:112]: Needs a newline

**Review by bcrochet — CHANGES_REQUESTED** [review]
Needs a couple of EOF newlines

File: openshift-gitops-operator/templates/argocd.yml.j2 (hunk 6 above)
Comment (acmenezes) [reply openshift-gitops-operator/templates/argocd.yml.j2]: Hi @bcrochet , I added the newline but just let it noted ansible-lint is not picking this up. Not sure if for Jinja files this is actually needed.

**Pull request merged**